File: LeetCode/LeetCode Practice Problems/1656- Design an Ordered Stream.cpp
class OrderedStream {
public:
    
    int id = 1;
    map<int,string> m;
    
    OrderedStream(int n) {
        
    }
    
    vector<string> insert(int idKey, string value) {
        m[idKey] = value;
        vector<string> ans;
        
        while(m.find(id) != m.end()) {
            ans.push_back(m[id]);
            id++;
        }
        
        return ans;        
    }
};

File: LeetCode/LeetCode Practice Problems/1656-Design an Ordered Stream.java
class OrderedStream {
    
    String st[];
    int n;
    int ptr = 0;

    public OrderedStream(int n) {
        this.n = n;
        st = new String[n];
    }
    
    public List<String> insert(int idKey, String value) {
        List<String> ans = new ArrayList();
        
        st[idKey - 1] = value;
       
        for(;ptr<n;ptr++) {
            if(st[ptr] == null) break;
            ans.add(st[ptr]);
        }
        
        return ans;
    }
}
